src/ng-publish.ts

```diff
@@ -113,6 +123,22 @@ export async function ngPublish(projectInfo: ProjectInfo): Promise<void> {
   await thisModule.pushTmpRepo(repositoryUrl, tag, tmpRepoDir);
 }
 
+export async function prePublishToGit(prePublishToGitCommand: string): Promise<Readable> {
+  const consoleOutput = chalk.dim('Executing prePublishiToGit with command: ') + prePublishToGitCommand;
+  console.log(consoleOutput);
+
+  const { stdout } = await exec(prePublishToGitCommand,
```

Comment:
'exec' doesn't return a promise and await does nothing here. 'stdout' is a regular value of type Readable and it is returned as a promise here, which does nothing but resolve immediately. The caller of this function will get a resolved promise before the exec command is finished. 

Exec is a function with a callback, you need to either convert the function using promisify or return a proper Promise yourself (see execProcessVerbose for an example of the later).